api/v1beta1/placementapi_webhook.go

```diff
@@ -86,7 +92,25 @@ func (r *PlacementAPI) ValidateCreate() error {
 func (r *PlacementAPI) ValidateUpdate(old runtime.Object) error {
 	placementapilog.Info("validate update", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object update.
+	oldPlacement, ok := old.(*PlacementAPI)
+	if !ok || oldPlacement == nil {
+		return apierrors.NewInternalError(fmt.Errorf("unable to convert existing object"))
+	}
+
+	if ! r.Spec.AdvancedMode {
+		if r.Spec.DatabaseInstance != oldPlacement.Spec.DatabaseInstance {
+			return apierrors.NewForbidden(
+				schema.GroupResource{
+					Group:    GroupVersion.WithKind("PlacementAPI").Group,
+					Resource: GroupVersion.WithKind("PlacementAPI").Kind,
+				}, r.GetName(), field.Forbidden(
+					field.NewPath("spec").Child("databaseInstance"),
+					"Value is immutable.",
+				),
+			)
+		}
+	}
```

Comment:
This logic needs to be in a function that is callable from openstack-operator validation webhook. The pattern is documented in https://github.com/openstack-k8s-operators/docs/blob/main/webhooks.md#reusing-webhook-logic